src/Microsoft.Azure.WebJobs.ServiceBus/Config/ServiceBusExtensionConfigProvider.cs

```diff
@@ -22,25 +22,25 @@ namespace Microsoft.Azure.WebJobs.ServiceBus.Config
     internal class ServiceBusExtensionConfigProvider : IExtensionConfigProvider
     {
         private readonly INameResolver _nameResolver;
-        private readonly IConnectionStringProvider _connectionStringProvider;
+        private readonly IConfiguration _configuration;
         private readonly ILoggerFactory _loggerFactory;
-        private readonly ServiceBusOptions _serviceBusOptions;
+        private readonly ServiceBusOptions _options;
         private readonly MessagingProvider _messagingProvider;
 
         /// <summary>
         /// Creates a new <see cref="ServiceBusExtensionConfigProvider"/> instance.
         /// </summary>
-        /// <param name="serviceBusConfig">The <see cref="ServiceBusOptions"></see> to use./></param>
-        public ServiceBusExtensionConfigProvider(IOptions<ServiceBusOptions> serviceBusConfig,
+        /// <param name="options">The <see cref="ServiceBusOptions"></see> to use./></param>
+        public ServiceBusExtensionConfigProvider(IOptions<ServiceBusOptions> options,
             MessagingProvider messagingProvider,
             INameResolver nameResolver,
-            IConnectionStringProvider connectionStringProvider,
+            IConfiguration configuration,
```

Comment:
Would be nice to avoid having to depend on the whole configuration here. Ideally, the options instance we take in would come fully populated, with connection strings and all.